src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -105,16 +153,6 @@ export default abstract class ComparisonStore {
                     !!this.tabHasBeenShown.get(GroupComparisonTab.SURVIVAL) ||
                         this.showSurvivalTab
                 );
-                this.tabHasBeenShown.set(
```

Comment:
you need to add an entry here, right? 

```
this.tabHasBeenShown.set(
  GroupComparisonTab.ALTERATIONS
  ...
)
```

Reply:
Added. What does this reaction do?

Reply:
tabs are shown based on available data, which can vary based on which groups are selected in the comparison session. but we determined it to be bad UX for a tab to disappear. so what we do is if, at any time, a tab is shown, then it will continue to be shown but possibly with a "no data" message no matter what happens next. actually it would be great to have this in a comment. would you mind putting it in while you're here?

Reply:
Ok, thank you for explaining.

Reply:
@adamabeshouse I did not read until the end of your comment. I have added the following comment:
>// When groups in the comparison are updated by the user
// certain tabs that were visible before might no longer be
// supported by the data and would be hidden. Disappearing
// tabs without explanation is considered as bad UX design.
// The logic below keeps track of tabs that were shown before
// and keeps them visible between groups updates.